src/components/ChatBox.tsx

```diff
@@ -0,0 +1,278 @@
+
+interface ChatBoxProperties {
+  recipientEmail: string;
+  profilePicture: string | null;
+  onClose: Function;
+  onMinimize: Function;
+  name: string;
+  surname: string;
+  profile: string;
+}
+
+const ChatBox: React.FC<ChatBoxProperties> = ({
+  recipientEmail,
+  profilePicture,
+  onClose,
+  onMinimize,
+  name,
+  surname,
+  profile,
+}) => {
+  const [wsClient, setWsClient] = useState<Client | null>(null);
+  const [doesChatRoomExist, setDoesChatRoomExist] = useState<boolean | null>(
+    null
+  );
+  const wsClientRef = useRef(wsClient);
+  const [chatId, setChatId] = useState<number | null>(null);
+  const [messages, setMessages] = useState<ChatMessage[]>([]);
+  const messagesRef = useRef(messages);
+  const [lastSeenMessage, setLastSeenMessage] = useState<number | undefined>();
+  const { timeZone } = Intl.DateTimeFormat().resolvedOptions();
+
+  useEffect(() => {
+    wsClientRef.current = wsClient;
+  }, [wsClient]);
+  useEffect(() => {
+    initWebsocketConnection();
+    return () => {
+      disconnectWebSocket();
+    };
+  }, [recipientEmail]); //to change, when chat with other recipient selected
+
+  const initWebsocketConnection = () => {
+    const socket = new sockjs(
+      `http://localhost:8081/ws?token=${store.user.jwtToken}`
+    );
+    const client = Stomp.over(() => socket);
+    const onConnect = (frame: IFrame) => {
+      setWsClient(client);
+      console.log("WS client connected");
+    };
+    client.connect({}, onConnect);
+  };
+
+  useEffect(() => {
+    if (
+      wsClient !== null &&
+      store.user.profile !== null &&
+      store.user.profile?.selected_profile !== null
+    ) {
+      checkIfChatRoomExists();
+    }
+  }, [wsClient, store.user.profile?.selected_profile]);
+
+  const checkIfChatRoomExists = async () => {
+    try {
+      const data = await api.getChatRoomWithGivenUser(recipientEmail, timeZone);
+      setChatId(data.id);
+      setDoesChatRoomExist(true);
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        const regex = /Status code: ([0-9]{3})/;
+        const match = error.message.match(regex);
+        if (match != null && match[1] === "404") {
+          setDoesChatRoomExist(false);
+        }
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (doesChatRoomExist == true && chatId != null) {
+      getMessages();
+    }
+  }, [doesChatRoomExist, chatId]);
+
+  useEffect(() => {
+    messagesRef.current = messages;
+  }, [messages]);
+
+  useEffect(() => {
+    console.log(`chatId: ${chatId}`);
+    if (chatId != null) {
+      subscribeToChatRoom();
+    }
+  }, [chatId]);
+
+  const subscribeToChatRoom = () => {
+    if (wsClient === null) {
+      console.log("not subscribing to chat room");
+      return;
+    }
+    const headers = {
+      "Accept-Role":
+        store.user.profile?.selected_profile?.toUpperCase() as string,
+      "Accept-Timezone": timeZone, //TODO: change for correct timezone
```

Comment:
Tutaj możesz użyć tego czego ja też używam w przy powiadomieniach:
```suggestion
      "Accept-Timezone": Intl.DateTimeFormat().resolvedOptions().timeZone
```

Źródło: https://stackoverflow.com/a/34602679

Reply:
A dobra, widzę że chyba tego używasz, tylko komentarz w kodzie nieaktualny.

Reply:
Taak, zapomniałem o nim